corehq/apps/accounting/tests/test_customer_invoicing.py

```diff
@@ -0,0 +1,389 @@
+        subscription_start_date = datetime.date(2016, 2, 23)
+        subscription_end_date = add_months_to_date(subscription_start_date, cls.subscription_length)
+        cls.subscription = generator.generate_domain_subscription(
+            cls.account,
+            cls.domain,
+            date_start=subscription_start_date,
+            date_end=subscription_end_date,
+        )
+        cls.subscription.plan_version.plan.is_customer_software_plan = True
+
+        advanced_subscription_end_date = add_months_to_date(subscription_end_date, 2)
+        cls.domain2 = generator.arbitrary_domain()
+        cls.sub2 = generator.generate_domain_subscription(
+            cls.account,
+            cls.domain2,
+            date_start=subscription_start_date,
+            date_end=advanced_subscription_end_date,
+            plan_version=cls.advanced_plan
+        )
+        cls.sub2.plan_version.plan.is_customer_software_plan = True
+
+        cls.domain3 = generator.arbitrary_domain()
+        cls.sub3 = generator.generate_domain_subscription(
+            cls.account,
+            cls.domain3,
+            date_start=subscription_start_date,
+            date_end=advanced_subscription_end_date,
+            plan_version=cls.advanced_plan
+        )
+        cls.sub3.plan_version.plan.is_customer_software_plan = True
+
+    def tearDown(self):
+        for user in self.domain.all_users():
+            user.delete()
+
+        for user in self.domain2.all_users():
+            user.delete()
+
+        for user in self.domain3.all_users():
+            user.delete()
+
+        super(BaseAccountingTest, self).tearDown()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.domain.delete()
+        cls.domain2.delete()
+        cls.domain3.delete()
+
+        super(BaseCustomerInvoiceCase, cls).tearDownClass()
+
+
+class TestCustomerInvoice(BaseCustomerInvoiceCase):
+
+    def test_multiple_subscription_invoice(self):
+        invoice_date = utils.months_from_date(self.subscription.date_start,
+                                              random.randint(2, self.subscription_length))
+        invoice_start, invoice_end = get_previous_month_date_range(invoice_date)
+        invoice_factory = CustomerAccountInvoiceFactory(
+            account=self.account,
+            date_start=invoice_start,
+            date_end=invoice_end
+        )
+        invoice_factory.create_invoice()
+
+        # The customer invoice will be added to one of the subscriptions invoice_set
+        invoice_set = self.subscription.invoice_set.union(self.sub2.invoice_set.all(),
+                                                          self.sub3.invoice_set.all())
+        self.assertEqual(invoice_set.count(), 1)
+
+        invoice = invoice_set.first()
+        self.assertGreater(invoice.balance, Decimal('0.0000'))
+        self.assertEqual(invoice.account, self.account)
+        self.assertIn(invoice.subscription, [self.subscription, self.sub2, self.sub3])
```

Comment:
I'd recommend that we allow ```invoice.subscription``` to be ```None``` when ```invoice.subscription.is_customer_account``` is ```True```.   The reason being is that it doesn't make sense to assign ```invoice.subscription``` to one subscription over another.

We can create a SQL constraint to ensure that we don't violate this condition.  I'm thinking of something like this:
https://github.com/dimagi/commcare-hq/blob/4f488551b0338b16c0428a7426ee9f097a84933c/corehq/apps/accounting/migrations/0006_unique_active_domain_subscription.py